File: cpp/1.cpp
#include <iostream>
#include <string> // Para usar std::getline()

using namespace std;

int main() {
    double comprimento, largura, perimetro;

    cout << "Digite o comprimento do retangulo ou lado do quadrado: ";
    cin >> comprimento;

    cout << "Digite a largura do retangulo (ou pressione Enter para usar o mesmo valor do comprimento): ";
    cin.ignore(); // Para limpar o buffer do teclado antes de usar getline()
    string larguraInput;
    getline(cin, larguraInput);

    if (larguraInput.empty()) {
        perimetro = comprimento * 4;
        cout << "O perimetro do quadrado e: " << perimetro << endl;
    } else {
        largura = stod(larguraInput); // Converter a string para double
        perimetro = 2 * (comprimento + largura);
        cout << "O perimetro do retangulo e: " << perimetro << endl;
    }

    return 0;
}

File: Lua/1.lua
io.write("Digite o comprimento do retângulo ou lado do quadrado: ")
    local comprimento = tonumber(io.read())
    
    io.write("Digite a largura do retângulo (ou pressione Enter para usar o mesmo valor do comprimento): ")
    local largura_input = io.read()
    
    local largura
    if largura_input == "" then
        largura = comprimento
    else
        largura = tonumber(largura_input)
    end
    
    local perimetro
    if largura == comprimento then
        perimetro = comprimento * 4
        print("O perímetro do quadrado é:", perimetro)
    else
        perimetro = 2 * (comprimento + largura)
        print("O perímetro do retângulo é:", perimetro)
    end
    